src/blob.rs

```diff
@@ -64,6 +64,9 @@ pub async fn download(progress: &impl Progress, source: &str, destination: &Path
     }
 
     let response = client.get(url).send().await?;
+    if !response.status().is_success() {
```

Comment:
I think you can just chain `error_from_status()` after the `await`?

Reply:
Yup, updated